delta-app/src/main/java/io/cdap/delta/store/StateStoreFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import io.cdap.cdap.api.worker.WorkerContext;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * This factory initiates the state store based on the new sql feature flag (which is environment dependent)
+ */
+public class StateStoreFactory {
+
+  public static final String DB_KEY = "SQL";
+  public static final String HCFS_KEY = "HCFS";
+
+  public StateStore getInstance(String stateStoreType, WorkerContext context, String offsetBasePath)
+    throws IOException {
+    if (stateStoreType != null && stateStoreType.toUpperCase(Locale.ROOT).trim().equals(DB_KEY)) {
```

Comment:
this is equivalent to `DB_KEY.equalsIgnoreCase(stateStoreType)`

Reply:
However, it's weird that somebody could put in a value like "asldfknnvz" and it would result in the HCFS store. 

It would be better to have an enum and pass that in here as the type. Validation of the type should occur in the app's configure() method.

Reply:
The idea was that for non-ODF environment, even if the CDC json is not populated, It will default to HCFS statestore.
But yeah I will make it ENUM , but default "NULL"  to HCFS .

Reply:
A workercontext is needed to access the cdc.json config since, it only available to App via REST calls. 
So, in configure method, there is no workercontext to access the system service. Hence in configure method, I donot have the value of the cdc.json config.

Reply:
I'm not sure I understand how the config is related to using an enum here. Can you describe what the problem is?